src/main/java/ui/UI.java

```diff
@@ -307,7 +307,7 @@ private void setupMainStage(Scene scene) {
         mainStage.setTitle("HubTurbo " + Utility.version(VERSION_MAJOR, VERSION_MINOR, VERSION_PATCH));
         mainStage.setScene(scene);
         mainStage.show();
-        mainStage.setOnCloseRequest(e -> quit());
+        mainStage.setOnCloseRequest(e -> quit(false));
```

Comment:
It may be better to make a new method for logging out instead, which does the cookie cleanup, then calls `quit`.